tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -136,6 +137,17 @@ public static void Initialize()
                 Log.Error(ex, ex.Message);
             }
 
+            try
+            {
+                // We need to initialize the RuntimeId class here so we can share the runtime id amongst products (profiler, tracer, debugger...)
+                RuntimeId.InitializeFromNative();
```

Comment:
Shouldn't you make that call lazily in `RuntimeId.Get()` instead? Unless I'm mistaken, `Instrumentation.Initialize` won't be call in child appdomains, so those domains will get an id generated with `Guid.NewGuid()`

Reply:
This is what I did first, @tonyredondo reminded me about `partial trust` applications and the fact we will make P/Invoke calls.

> Unless I'm mistaken, Instrumentation.Initialize won't be call in child appdomains, so those domains will get an id generated with Guid.NewGuid()

🤔 ok, did not know that. 
If we can catch the exception in `partial trust` case, we could do it in `RuntimeId.Get()`

Reply:
I'm checking the docs, maybe we can use this [property](https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.isfullytrusted?redirectedfrom=MSDN&view=net-6.0#System_AppDomain_IsFullyTrusted) in the `RuntimeId.Get()` to check if we are allowed to do PInvoke calls.

Reply:
https://github.com/DataDog/dd-trace-dotnet/pull/2474/commits/ecb34d13642094717ad2e65e3fa0d189b80d1b06